tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -979,3 +1007,150 @@ def _get_num_gpus_on_instance(instance_type_info):
     }
     """
     return sum([gpu_type.get("Count") for gpu_type in instance_type_info.get("GpuInfo").get("Gpus")])
+
+
+@retry(wait_fixed=seconds(20), stop_max_delay=minutes(5))
+def _wait_for_computefleet_changed(cluster, desired_status):
+    assert_that(cluster.status()).contains(f"ComputeFleetStatus: {desired_status}")
+
+
+@retry(wait_fixed=seconds(20), stop_max_delay=minutes(5))
+def _wait_for_partition_state_changed(scheduler_commands, partition, desired_state):
+    assert_that(scheduler_commands.get_partition_state(partition=partition)).is_equal_to(desired_state)
+
+
+def _update_and_start_cluster(cluster):
+    cluster.stop()
+    _wait_for_computefleet_changed(cluster, "STOPPED")
+    cluster.update()
+    cluster.start()
+    _wait_for_computefleet_changed(cluster, "RUNNING")
+
+
+def _inject_bootstrap_failures(cluster, bucket_name, pcluster_config_reader):
+    """Update cluster to include pre-install script, which introduce bootstrap error."""
+    updated_config_file = pcluster_config_reader(config_file="pcluster.config.update.ini", bucket=bucket_name)
```

Comment:
Very minor: since you have `pcluster.config.recover.ini`, it might be worth to call this config used to break the cluster `pcluster.config.broken.ini`?

Reply:
Renamed